pkg/api/controllers/query.go

```diff
@@ -0,0 +1,38 @@
+package controllers
+
+import (
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/numary/ledger/pkg/ledger"
+	"github.com/numary/ledger/pkg/storage"
+)
+
+const (
+	MaxPageSize     = 1000
+	DefaultPageSize = storage.QueryDefaultPageSize
+)
+
+var (
+	ErrInvalidPageSize = ledger.NewValidationError("invalid query value 'page_size'")
+)
+
+func getPageSize(c *gin.Context) (uint, error) {
+	var (
+		pageSize uint64
+		err      error
+	)
+	if pageSizeParam := c.Query("page_size"); pageSizeParam != "" {
+		pageSize, err = strconv.ParseUint(pageSizeParam, 10, 64)
+		if err != nil {
+			return 0, ErrInvalidPageSize
+		}
+
+		if pageSize > MaxPageSize {
+			pageSize = MaxPageSize
+		}
+	} else {
```

Comment:
maybe we can print a log here ?

Reply:
Why? It we print a log each time the code take a default value, it will give a lot of useless logs.

Reply:
Also, i don't remind if it complete, i have to check, but all requests should be logged using opentelemetry.

Reply:
Why? If we log each time a default value is used, it will give a lot of logs ^^

Reply:
i didn't mean especially logging the default value, but logging that the parameter given was higher than expeced. I think this can be useful to determine how our api (and our parameters) are used by our customers in way we didn't predict ?

Reply:
Ok, i agree.
Instead of a log, i think an opentelemetry attributes is well suited.

Reply:
@jdupas22 I checked the instrumentation library used to handle http request (https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin) and it register the full path inside an attribute named "http.target".